hw2-9

FeralBerry opened the pull request.

FeralBerry committed:
hw2-9

dmitry-bizin commented on .idea/uiDesigner.xml:
папку .idea надо игнорировать

dmitry-bizin commented on src/main/java/org/example/Employee.java:
```diff
@@ -0,0 +1,41 @@
+    }
+    @Override
+    public boolean equals(Object other) {
+        if (this.getClass() != other.getClass()) {
```
нет проверки на null - не по контракту реализован equals

dmitry-bizin commented on src/main/java/org/example/Exceptions/EmployeeNotFoundException.java:
```diff
@@ -0,0 +1,11 @@
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ResponseStatus(HttpStatus.BAD_REQUEST)
```
```suggestion
@ResponseStatus(HttpStatus.NOT_FOUND)
```

dmitry-bizin commented on src/main/java/org/example/RestControllers/Departments.java:
```diff
@@ -0,0 +1,31 @@
+        this.departmentsService = departmentsService;
+    }
+    @GetMapping("/max-salary")
+    public float maxSalary(@RequestParam(name = "departmentId") int departmentId) {
```
надо возвращать сотрудника, а не число

dmitry-bizin commented on src/main/java/org/example/RestControllers/Departments.java:
```diff
@@ -0,0 +1,31 @@
+        return departmentsService.maxSalary(departmentId);
+    }
+    @GetMapping("/min-salary")
+    public float minSalary(@RequestParam(name = "departmentId") int departmentId) {
```
надо возвращать сотрудника, а не число

dmitry-bizin commented on src/main/java/org/example/RestControllers/Departments.java:
```diff
@@ -0,0 +1,31 @@
+        return departmentsService.minSalary(departmentId);
+    }
+    @GetMapping(path = "/all")
+    public String all(@RequestParam(name = "departmentId",required = false) String departmentId) {
```
надо возвращать сотрудников, а не строку. Также методов должно быть 2, а не 1

dmitry-bizin commented on src/main/java/org/example/RestControllers/Departments.java:
```diff
@@ -0,0 +1,31 @@
+        if(departmentId == null){
+            return departmentsService.allDepartments();
+        }
+        return departmentsService.all(Integer.parseInt(departmentId));
```
нет смысла парсить строку, когда можно сразу принять число

dmitry-bizin commented on src/main/java/org/example/RestControllers/Employee.java:
```diff
@@ -0,0 +1,35 @@
+        this.employeeService = employeeService;
+    }
+    @GetMapping("/add")
+    public String add(@RequestParam(value = "firstName", required = false) String firstName,
```
метод должен возвращать сотрудника, а не строку

FeralBerry replied:
объект сотрудника? Просто в данном случае метод возвращает строку со всеми данными сотрудника ФИО департамент и зарплату

dmitry-bizin replied:
Да, а должен возвращаться JSON, про это писалось ещё в прошлой домашке. Также я на разборе про это рассказывал, рекомендую посещать или хотя бы смотреть в записи

FeralBerry replied:
ок, сейчас тогда сделаю JSON там пару строк переделать

dmitry-bizin replied:
переделывать ничего не нужно, надо просто вернуть объект сотрудника, а не строку, как я и написал

dmitry-bizin commented on src/main/java/org/example/RestControllers/Employee.java:
```diff
@@ -0,0 +1,35 @@
+    }
+    @GetMapping("/add")
+    public String add(@RequestParam(value = "firstName", required = false) String firstName,
+                      @RequestParam(value = "lastName", required = false) String lastName,
+                      @RequestParam(value = "departmentId", required = false) int departmentId,
+                      @RequestParam(value = "salary", required = false) float salary) {
```
почему параметры не обязательные?

dmitry-bizin commented on src/main/java/org/example/RestControllers/Employee.java:
```diff
@@ -0,0 +1,35 @@
+        return employeeService.add(firstName,lastName,departmentId,salary);
+    }
+    @GetMapping("/remove")
+    public String remove(@RequestParam (value = "firstName", required = false) String firstName,
```
метод должен возвращать сотрудника, а не строку

dmitry-bizin commented on src/main/java/org/example/RestControllers/Employee.java:
```diff
@@ -0,0 +1,35 @@
+        return employeeService.remove(firstName,lastName);
+    }
+    @GetMapping("/find")
+    public String find(@RequestParam (value = "firstName", required = false) String firstName,
```
метод должен возвращать сотрудника, а не строку

dmitry-bizin commented on src/main/java/org/example/RestControllers/Employee.java:
```diff
@@ -0,0 +1,35 @@
+    }
+    @GetMapping("/remove")
+    public String remove(@RequestParam (value = "firstName", required = false) String firstName,
+                         @RequestParam(value = "lastName", required = false) String lastName) {
```
почему параметры не обязательные?

dmitry-bizin commented on src/main/java/org/example/RestControllers/Employee.java:
```diff
@@ -0,0 +1,35 @@
+    }
+    @GetMapping("/find")
+    public String find(@RequestParam (value = "firstName", required = false) String firstName,
+                       @RequestParam(value = "lastName", required = false) String lastName) {
```
почему параметры не обязательные?

dmitry-bizin commented on src/main/java/org/example/Services/DepartmentsService.java:
```diff
@@ -0,0 +1,72 @@
+    }
+
+    public float maxSalary(int departmentId){
+        float max = 0;
+        String[] str;
+        float str_parse;
+        for(int i = 0; i < searchByDepartment(departmentId).size(); i++){
+            str = searchByDepartment(departmentId).toString().split("salary: ");
+            str_parse = Float.parseFloat(removeLastChar(str[1]));
+            if(str_parse > max){
+                max = str_parse;
+            }
+        }
+        return max;
```
а где стримы? здесь код будет в одну строку через стрим

dmitry-bizin commented on src/main/java/org/example/Services/DepartmentsService.java:
```diff
@@ -0,0 +1,72 @@
+        return max;
+    }
+    public float minSalary(int departmentId){
+        float min = Integer.MAX_VALUE;
+        String[] str;
+        float str_parse;
+        if(searchByDepartment(departmentId).size() > 0){
+            for(int i = 0; i < searchByDepartment(departmentId).size(); i++){
+                str = searchByDepartment(departmentId).toString().split("salary: ");
+                str_parse = Float.parseFloat(removeLastChar(str[1]));
+                if(str_parse < min){
+                    min = str_parse;
+                }
+            }
+            return min;
+        } else {
+            return 0;
+        }
```
а где стримы? здесь код будет в одну строку через стрим

dmitry-bizin commented on src/main/java/org/example/Services/DepartmentsService.java:
```diff
@@ -0,0 +1,72 @@
+        }
+    }
+    public String all(int departmentId){
+        return searchByDepartment(departmentId).toString();
```
а где стримы?

dmitry-bizin commented on src/main/java/org/example/Services/DepartmentsService.java:
```diff
@@ -0,0 +1,72 @@
+        return searchByDepartment(departmentId).toString();
+    }
+    public String allDepartments(){
+        return searchByDepartment(0).toString();
```
а где стримы?

dmitry-bizin commented on src/main/java/org/example/Services/EmployeeService.java:
```diff
@@ -0,0 +1,57 @@
+        try{
+            validateEmployeeName(firstName,lastName);
+            employeesList.put(new Employee(firstName,lastName).toKey(),new Employee(firstName,lastName,departmentId,salary).toString());
+        } catch (EmployeeAlreadyAddedException e) {
```
не нужно ловить исключения, в этом как раз и смысл, что ловит их спринг и возвращает указанный код ответа, не надо захламлять код try/catch-ми

dmitry-bizin commented on src/main/java/org/example/Services/EmployeeService.java:
```diff
@@ -0,0 +1,57 @@
+        try {
+            validateSearch(firstName,lastName);
+            employeesList.remove(new Employee(firstName,lastName).toKey());
+        } catch (EmployeeNotFoundException e){
```
не нужно ловить исключения, в этом как раз и смысл, что ловит их спринг и возвращает указанный код ответа, не надо захламлять код try/catch-ми

dmitry-bizin commented on src/main/java/org/example/Services/EmployeeService.java:
```diff
@@ -0,0 +1,57 @@
+        try {
+            validateSearch(firstName,lastName);
+            employee = employeesList.get(new Employee(firstName, lastName).toKey());
+        }catch (EmployeeNotFoundException e){
```
не нужно ловить исключения, в этом как раз и смысл, что ловит их спринг и возвращает указанный код ответа, не надо захламлять код try/catch-ми

dmitry-bizin commented on src/main/java/org/example/Exceptions/EmployeeAlreadyAddedException.java:
```diff
@@ -0,0 +1,11 @@
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ResponseStatus(HttpStatus.BAD_REQUEST)
+public class EmployeeAlreadyAddedException extends Exception{
```
```suggestion
public class EmployeeAlreadyAddedException extends RuntimeException{
```

dmitry-bizin commented on src/main/java/org/example/Exceptions/EmployeeNotFoundException.java:
```diff
@@ -0,0 +1,11 @@
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ResponseStatus(HttpStatus.BAD_REQUEST)
+public class EmployeeNotFoundException extends Exception{
```
```suggestion
public class EmployeeNotFoundException extends RuntimeException{
```

FeralBerry committed:
Переделано на стримы

FeralBerry committed:
изменен гит игнор

FeralBerry committed:
почистил код

FeralBerry committed:
Переделал все методы возвращают JSON

dmitry-bizin commented on src/main/java/org/example/RestControllers/Departments.java:
```diff
@@ -14,18 +17,19 @@ public Departments(DepartmentsService departmentsService) {
     }
     @GetMapping("/max-salary")
-    public float maxSalary(@RequestParam(name = "departmentId") int departmentId) {
+    public List maxSalary(@RequestParam(name = "departmentId") int departmentId) {
```
почему лист? один сотрудник должен быть

dmitry-bizin commented on src/main/java/org/example/RestControllers/Departments.java:
```diff
@@ -14,18 +17,19 @@ public Departments(DepartmentsService departmentsService) {
     }
     @GetMapping("/min-salary")
-    public float minSalary(@RequestParam(name = "departmentId") int departmentId) {
+    public List minSalary(@RequestParam(name = "departmentId") int departmentId) {
```
почему лист? один сотрудник должен быть

dmitry-bizin commented on src/main/java/org/example/RestControllers/Departments.java:
```diff
@@ -14,18 +17,19 @@ public Departments(DepartmentsService departmentsService) {
     }
     @GetMapping(path = "/all")
-    public String all(@RequestParam(name = "departmentId",required = false) String departmentId) {
+    public List all(@RequestParam(name = "departmentId") String departmentId) {
```
List надо с дженерик-параметром писать

dmitry-bizin commented on src/main/java/org/example/RestControllers/Employee.java:
```diff
@@ -16,20 +18,20 @@ public Employee(EmployeeService employeeService) {
-                      @RequestParam(value = "lastName", required = false) String lastName,
-                      @RequestParam(value = "departmentId", required = false) int departmentId,
-                      @RequestParam(value = "salary", required = false) float salary) {
+    public Map add(@RequestParam(value = "firstName") String firstName,
```
почему Map? сотрудник должен быть добавленный

dmitry-bizin commented on src/main/java/org/example/RestControllers/Employee.java:
```diff
@@ -16,20 +18,20 @@ public Employee(EmployeeService employeeService) {
     @GetMapping("/remove")
-    public String remove(@RequestParam (value = "firstName", required = false) String firstName,
-                         @RequestParam(value = "lastName", required = false) String lastName) {
+    public String remove(@RequestParam (value = "firstName") String firstName,
```
надо вернуть сотрудника, а не строку

dmitry-bizin commented on src/main/java/org/example/RestControllers/Employee.java:
```diff
@@ -16,20 +18,20 @@ public Employee(EmployeeService employeeService) {
     @GetMapping("/find")
-    public String find(@RequestParam (value = "firstName", required = false) String firstName,
-                       @RequestParam(value = "lastName", required = false) String lastName) {
+    public String find(@RequestParam (value = "firstName") String firstName,
```
надо вернуть сотрудника, а не строку

dmitry-bizin commented on src/main/java/org/example/Services/DepartmentsService.java:
```diff
@@ -2,71 +2,61 @@
     private final EmployeeService employeeService;
 
-    public DepartmentsService(EmployeeService employeeService) {
+    private DepartmentsService(EmployeeService employeeService) {
```
работать не будет, конструктор должен быть публичный

FeralBerry replied:
Всё работает) программа протестирована и работает с приватным, потому что этот конструктор используется только внутри этого класса

dmitry-bizin replied:
кем он используется внутри класса? этот конструктор будет и должен использоваться только спрингом

BizinMitya commented on src/main/java/org/example/Services/DepartmentsService.java:
```diff
@@ -2,71 +2,61 @@
-            if(str_parse > max){
-                max = str_parse;
+    private List findEmployee(Map map, String keySearch){
+        List <Object> employee = new ArrayList<>();
```
что это такое? надо найти сотрудника и вернуть, что сложного?

dmitry-bizin commented on src/main/java/org/example/Services/DepartmentsService.java:
```diff
@@ -2,71 +2,61 @@
-        return max;
+        });
+        return employee;
     }
```
этот метод здесь не нужен

dmitry-bizin commented on src/main/java/org/example/Services/DepartmentsService.java:
```diff
@@ -2,71 +2,61 @@
-                    min = str_parse;
-                }
+    public List maxSalary(int departmentId) {
+        AtomicReference<Float> max = new AtomicReference<>((float) 0);
+        AtomicReference<String> keySearch = new AtomicReference<>("");
```
зачем это?

FeralBerry replied:
автоматически генерирует идея на 21 Java

dmitry-bizin replied:
надо самому писать код и думать, а не генерировать идеей.

dmitry-bizin commented on src/main/java/org/example/Services/DepartmentsService.java:
```diff
@@ -2,71 +2,61 @@
-    }
-    public String allDepartments(){
-        return searchByDepartment(0).toString();
+        });
```
надо использовать стримы

FeralBerry replied:
скрытый stream метод  .forEach можно менять на .stream().forEach работать будет и так и так

dmitry-bizin replied:
да, но решить надо в 1 строку через стрим, а не в 100500 строк

dmitry-bizin commented on src/main/java/org/example/Services/DepartmentsService.java:
```diff
@@ -2,71 +2,61 @@
+    public List minSalary(int departmentId) {
+        AtomicReference<Float> min = new AtomicReference<>((float) Integer.MAX_VALUE);
+        AtomicReference<String> keySearch = new AtomicReference<>("");
+        employeeService.getEmployees().forEach((key, value) -> {
```
надо использовать стримы

dmitry-bizin commented on src/main/java/org/example/Services/DepartmentsService.java:
```diff
@@ -2,71 +2,61 @@
+    }
+    public List all(int departmentId) {
+        List <Object> employee = new ArrayList<>();
+        employeeService.getEmployees().forEach((key, value) -> {
```
надо использовать стримы

dmitry-bizin commented on src/main/java/org/example/Services/DepartmentsService.java:
```diff
@@ -2,71 +2,61 @@
+    public List allDepartments() {
+        List <Object> employee = new ArrayList<>();
+        employeeService.getEmployees().forEach((key, value) ->
+            employee.add(value)
```
надо использовать стримы

dmitry-bizin commented on src/main/java/org/example/Services/EmployeeService.java:
```diff
@@ -11,47 +9,20 @@
-        return employeesList + "<br>";
+    public Map add(String firstName, String lastName,int departmentId,float salary) {
+        employeesList.put(new Employee(firstName,lastName).toKey(), new Employee(firstName,lastName,departmentId,salary));
+        return employeesList;
```
надо вернуть добалвенного сотрудника

dmitry-bizin commented on src/main/java/org/example/Services/EmployeeService.java:
```diff
@@ -11,47 +9,20 @@
-        }
-        return employeesList + "<br>";
+    public Map add(String firstName, String lastName,int departmentId,float salary) {
+        employeesList.put(new Employee(firstName,lastName).toKey(), new Employee(firstName,lastName,departmentId,salary));
```
зачем 2 раза создавать сотрудника?

dmitry-bizin commented on src/main/java/org/example/Services/EmployeeService.java:
```diff
@@ -11,47 +9,20 @@
-            System.out.println("Ошибка: " + e.getMessage());
-        }
+        employeesList.remove(new Employee(firstName,lastName).toKey());
         return employeesList.toString();
```
надо вернуть удаленного сотрудника

dmitry-bizin commented on src/main/java/org/example/Services/EmployeeService.java:
```diff
@@ -11,47 +9,20 @@
-            System.out.println("Ошибка: " + e.getMessage());
-        }
-        return employee;
+        return employeesList.get(new Employee(firstName, lastName).toKey()).toString();
```
надо вернуть найденного сотрудника

dmitry-bizin commented on src/main/java/org/example/Services/EmployeeService.java:
```diff
@@ -11,47 +9,20 @@
-    }
-    public Map<String,String> getEmployees(){
+    public Map<String,Employee> getEmployees(){
         return this.employeesList;
```
надо вернуть список, а не карту

FeralBerry committed:
Добавил комментарии что и где делается и что используется.
Переделал по требованию вывод сотрудника объектом класса Employee. Добавил обработку маршрута с параметром по подсказке

FeralBerry committed:
добавлены вторые реализации поиска по максимальной и минимальной зарплате всё с комментариями.

FeralBerry committed:
Сделано 3 реализации для этого задания все 3 удовлетворяют параметрам задания.

FeralBerry committed:
разбил реализации для читаемости

dmitry-bizin commented on src/main/java/org/example/RestControllers/Departments.java:
```diff
@@ -16,20 +15,25 @@ public class Departments {
+    // про такой вариант параметров не знал думал будет работать только с /{departmentId}
+    // Переписаны контроллер и сервис, которые возвращают всех сотрудников с разделением по отделам.
+    @GetMapping(value = "/all", params = {"departmentId"})
+    public List<Object> all(@RequestParam(name = "departmentId") int departmentId) {
```
```suggestion
    public List<Employee> all(@RequestParam(name = "departmentId") int departmentId) {
```

dmitry-bizin commented on src/main/java/org/example/RestControllers/Departments.java:
```diff
@@ -16,20 +15,25 @@ public class Departments {
+    }
+    // Переписаны контроллер и сервис, которые возвращают всех сотрудников по отделу
+    @GetMapping(value = "/all")
+    public List<Object> all() {
```
```suggestion
    public List<Employee> all() {
```

dmitry-bizin commented on src/main/java/org/example/RestControllers/Departments.java:
```diff
@@ -16,20 +15,25 @@ public class Departments {
     }
+    // Переписаны контроллер и сервис, которые возвращают сотрудника с минимальной зарплатой на основе номера отдела.
     @GetMapping("/min-salary")
     public List minSalary(@RequestParam(name = "departmentId") int departmentId) {
```
```suggestion
    public List<Employee> minSalary(@RequestParam(name = "departmentId") int departmentId) {
```

dmitry-bizin commented on src/main/java/org/example/RestControllers/Departments.java:
```diff
@@ -16,20 +15,25 @@ public class Departments {
     }
+    // Переписаны контроллер и сервис, которые возвращают сотрудника с максимальной зарплатой на основе номера отдела, который приходит в запрос из браузера.
     @GetMapping("/max-salary")
     public List maxSalary(@RequestParam(name = "departmentId") int departmentId) {
```
```suggestion
    public List<Employee> maxSalary(@RequestParam(name = "departmentId") int departmentId) {
```

dmitry-bizin commented on src/main/java/org/example/RestControllers/Employee.java:
```diff
@@ -18,20 +18,20 @@ public Employee(EmployeeService employeeService) {
-                   @RequestParam(value = "lastName") String lastName,
-                   @RequestParam(value = "departmentId") int departmentId,
-                   @RequestParam(value = "salary") float salary) {
+    public List<Object> add(@RequestParam(value = "firstName") String firstName,
```
```suggestion
    public List<Employee> add(@RequestParam(value = "firstName") String firstName,
```

dmitry-bizin commented on src/main/java/org/example/Services/DepartmentsService.java:
```diff
@@ -1,62 +1,172 @@
+        return employee1;
+    }*/
+    /* 2 реализация с компаратором
     public List maxSalary(int departmentId) {
```
```suggestion
    public List<Employee> maxSalary(int departmentId) {
```

dmitry-bizin commented on src/main/java/org/example/Services/DepartmentsService.java:
```diff
@@ -1,62 +1,172 @@
+        // return employee2;
+    }*/
+    // реализация 3 через стрим и компаратор, фильтр стрима возвращает объект сотрудника
+    public List maxSalary(int departmentId) {
```
```suggestion
    public List<Employee> maxSalary(int departmentId) {
```

dmitry-bizin commented on src/main/java/org/example/Services/DepartmentsService.java:
```diff
@@ -1,62 +1,172 @@
+                .filter(item -> item.getValue().getDepartment() == departmentId)
+                .max(Comparator.comparingDouble(item -> item.getValue().getSalary()))
+                .stream()
+                .collect(Collectors.toList());
```
в итоге всё равно будет один сотрудник только в списке) потому что max возвращает одного, поэтому делать список нет никакого смысла здесь. Или перепиши так, чтобы именно список возвращался из нескольких, а не одного

FeralBerry replied:
Спасибо! То что может быть несколько подумал, а что min и max возвращают 1 не подумал)

dmitry-bizin commented on src/main/java/org/example/Services/DepartmentsService.java:
```diff
@@ -1,62 +1,172 @@
+    public List minSalary(int departmentId) {
+        // можно задавать float min, но при работе 2 и более людей могут быть ошибки
+        AtomicReference<Float> min = new AtomicReference<>(0.0f);
+        List <Object> employee1 = new ArrayList<>();
```
```suggestion
        List <Employee> employee1 = new ArrayList<>();
```

dmitry-bizin commented on src/main/java/org/example/Services/DepartmentsService.java:
```diff
@@ -1,62 +1,172 @@
         });
-        return findEmployee(employeeService.getEmployees(),keySearch.get());
+        Optional<Float> minOptional = employeeSalary.stream().min(comparator);
+        List <Object> employee2 = new ArrayList<>();
```
```suggestion
        List <Employee> employee2 = new ArrayList<>();
```

dmitry-bizin commented on src/main/java/org/example/Services/DepartmentsService.java:
```diff
@@ -1,62 +1,172 @@
+        return employee2;
+    }*/
+    // реализация 3 через стрим и компаратор возвращает объект сотрудника
+    public List minSalary(int departmentId) {
```
```suggestion
    public List<Employee> minSalary(int departmentId) {
```

dmitry-bizin commented on src/main/java/org/example/Services/DepartmentsService.java:
```diff
@@ -1,62 +1,172 @@
+                .filter(item -> item.getValue().getDepartment() == departmentId)
+                .min(Comparator.comparingDouble(item -> item.getValue().getSalary()))
+                .stream()
+                .collect(Collectors.toList());
```
здесь как выше про список

dmitry-bizin commented on src/main/java/org/example/Services/DepartmentsService.java:
```diff
@@ -1,62 +1,172 @@
-    public List all(int departmentId) {
+    // Переписаны контроллер и сервис, которые возвращают всех сотрудников по отделу
+    /* реализация 1
+    public List<Object> allForDepartments(int departmentId) {
```
```suggestion
    public List<Employee> allForDepartments(int departmentId) {
```

dmitry-bizin commented on src/main/java/org/example/Services/DepartmentsService.java:
```diff
@@ -1,62 +1,172 @@
+    /* реализация 1
+    public List<Object> allForDepartments(int departmentId) {
+        // Объявляем пустую коллекцию
         List <Object> employee = new ArrayList<>();
```
```suggestion
        List <Employee> employee = new ArrayList<>();
```

dmitry-bizin commented on src/main/java/org/example/Services/DepartmentsService.java:
```diff
@@ -1,62 +1,172 @@
+        // Возвращаем коллекцию всех сотрудников в 1 отделе
+         return employee;
+    }*/
+    public List<Object> allForDepartments(int departmentId) {
```
```suggestion
    public List<Employee> allForDepartments(int departmentId) {
```

dmitry-bizin commented on src/main/java/org/example/Services/DepartmentsService.java:
```diff
@@ -1,62 +1,172 @@
+    public List<Object> all() {
+        return employeeService.getEmployees();
+    }*/
+    public List<Object> all() {
```
```suggestion
    public List<Employee> all() {
```

dmitry-bizin commented on src/main/java/org/example/Services/DepartmentsService.java:
```diff
@@ -1,62 +1,172 @@
+    }*/
+    public List<Object> all() {
+        // 2 вариант сохранение всех в коллекцию для вывода без ключей
         List <Object> employee = new ArrayList<>();
```
```suggestion
        List <Employee> employee = new ArrayList<>();
```

dmitry-bizin commented on src/main/java/org/example/Services/DepartmentsService.java:
```diff
@@ -1,62 +1,172 @@
+        // коллекцию employee сотрудника
+        // Реализация проекта заменена через циклы на Stream API.
         employeeService.getEmployees().forEach((key, value) ->
-            employee.add(value)
+                employee.add(value)
         );
```
нет никакой группировки по отделам

dmitry-bizin commented on src/main/java/org/example/Services/EmployeeService.java:
```diff
@@ -3,24 +3,38 @@
-    public Map add(String firstName, String lastName,int departmentId,float salary) {
-        employeesList.put(new Employee(firstName,lastName).toKey(), new Employee(firstName,lastName,departmentId,salary));
-        return employeesList;
+    public List<Object> add(String firstName, String lastName, int departmentId, float salary) {
```
```suggestion
    public List<Employee> add(String firstName, String lastName, int departmentId, float salary) {
```

FeralBerry committed:
разбил реализации для читаемости
